### PR TITLE
[JUJU-790] Add GitHub Action to run `go test`

### PR DESCRIPTION
**Pull request by barrettj12**
This PR introduces a GitHub Action which installs the version of Go specified in `go.mod`, then runs `go test`. This action will be run as CI on pushes and pull requests.

The action has been tested locally on my machine using nektos/act.

### CODE REVIEW

**Comment by barrettj12** [comment]
After fixing a small typo, it also [appears to succeed on GitHub](https://github.com/barrettj12/errors/runs/5651982166).

File: .github/workflows/gotest.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
+name: Run `go test`
+on: [push, pull_request, workflow_dispatch]
+jobs:
+  run-go-test:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout repo
+        uses: actions/checkout@v3
+      - name: Get version from go.mod
```
Comment (tlm) [inline .github/workflows/gotest.yml:9]: Can we get the go version the same way that Juju does just to keep it consistent? 

https://github.com/juju/juju/blob/0881dffdd57e535c5cf56e59c9fc897b74411438/.github/workflows/client-tests.yml#L40

Save us from having to trust a small action repo that we don't have much control over.

**Review by tlm — APPROVED** [review]
lgtm

File: .github/workflows/gotest.yml (hunk 1 above)
Comment (barrettj12) [reply .github/workflows/gotest.yml]: Done

**Pull request merged**